app/uk/gov/hmrc/pensionschemereturn/audit/ApiAuditUtil.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pensionschemereturn.audit
+
+import uk.gov.hmrc.pensionschemereturn.models.response.PsrSubmissionEtmpResponse
+import uk.gov.hmrc.pensionschemereturn.services.AuditService
+import play.api.mvc.RequestHeader
+import com.google.inject.Inject
+import play.api.http.Status
+import play.api.libs.json._
+import uk.gov.hmrc.http.{HttpException, HttpResponse, UpstreamErrorResponse}
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success, Try}
+
+class ApiAuditUtil @Inject()(auditService: AuditService) {
+
+  def firePsrPostAuditEvent(
+    pstr: String,
+    data: JsValue
+  )(implicit ec: ExecutionContext, request: RequestHeader): PartialFunction[Try[HttpResponse], Unit] = {
+    case Success(httpResponse) =>
```

Comment:
Would it be useful to log when the audit event happened? As there are in some cases delays between when the event is fired and when it appears in CIP.
We could disable/remove logs later if not needed.

Reply:
If you don't need the payload, above would be sufficient. Otherwise it is easy to add if you find they are needed.